### PR TITLE
clear up Pasta negate functions

### PR DESCRIPTION
**Pull request by mpenciak**
This PR adds distinct `negate` functions for base/scalar field elements over Pallas and Vesta curves

### CODE REVIEW

File: src/pasta/Vesta.sol [hunk 1]
```diff
@@ -126,8 +126,13 @@ library Vesta {
         return VestaProjectivePoint(p.x, P_MOD - (p.y % P_MOD), p.z);
     }
 
-    /// @return res = -fr the negation of scalar field element.
+    /// @return res = -fr the negation of base field element.
     function negate(uint256 fr) internal pure returns (uint256 res) {
```
Comment (storojs72) [inline src/pasta/Vesta.sol:130]: ditto

File: src/pasta/Pallas.sol [hunk 2]
```diff
@@ -126,8 +126,13 @@ library Pallas {
         return PallasProjectivePoint(p.x, P_MOD - (p.y % P_MOD), p.z);
     }
 
-    /// @return res = -fr the negation of scalar field element.
+    /// @return res = -fr the negation of base field element.
     function negate(uint256 fr) internal pure returns (uint256 res) {
```
Comment (storojs72) [inline src/pasta/Pallas.sol:130]: What about using two explicit function names: `negateBase` and `negateScalar`?

**Review by storojs72 — COMMENTED** [review]
I appreciate this change. At the same time it arises following questions:
- How to distinguish whether it is a base or scalar field element, having just `uint256` value in Solidity? Can we include this testing as a part of the https://github.com/lurk-lab/solidity-verifier/issues/11, @huitseeker? 
- Do we use in Nova only base field or scalar field elements, or their mix? Is it defined formally in paper, when this or that particular low-level algorithm is involved?
- Can we have additional unit tests that just ensure compatibility with Rust reference implementation of base field elements negation, @mpenciak ?

**Comment by mpenciak** [comment]
As discussed earlier, I think the best temporary solution may be the low tech solution of simply prefacing the `uint256` variables with the name of the curve for which they are the **base** field.

Though @huitseeker or others may have better solutions.

**Review by huitseeker — APPROVED** [review]
> As discussed earlier, I think the best temporary solution may be the low tech solution of simply prefacing the uint256 variables with the name of the curve for which they are the base field.

There's no definable type inference u256 -> {pallas,vesta}::{Base, Scalar}. I agree the best we can do is to signal intent with comments or variable names.
Note we'd rather want to communicate what a number is meant to be used as rather than what is is. So for a Scalar, you'd say "PallasScalar" rather than "VestaBase", even if both are true.

**Review by storojs72 — APPROVED** [review]

**Pull request merged**